### PR TITLE
rehydrate since

### PR DESCRIPTION
**Pull request by michaeldjeffrey**
- rehydrate config from dets files upon startup
- when starting the config update stream, send the timestamp of the last update we recieved

- todo: if any of the config cannot be rehydrated, remove everything and start from the beginning.

### CODE REVIEW

File: src/grpc/iot_config/hpr_eui_pair_storage.erl [hunk 1]
```diff
@@ -138,11 +153,42 @@ count_for_route(RouteID) ->
...
+    end).
+
+with_open_dets(FN) ->
+    DataDir = hpr_utils:base_data_dir(),
+    DETSFile = filename:join([DataDir, "hpr_eui_pair_storage.dets"]),
+    ok = filelib:ensure_dir(DETSFile),
+
+    case
+        dets:open_file(?DETS, [
+            {file, DETSFile}, {type, bag}, {keypos, hpr_route_ets:ets_keypos()}
```
Comment (michaeldjeffrey) [inline src/grpc/iot_config/hpr_eui_pair_storage.erl:183]: This needs to be eui_pair_storage key pos

File: src/grpc/iot_config/hpr_route_storage.erl [hunk 2]
```diff
@@ -124,19 +143,40 @@ test_size() ->
...
 %% -------------------------------------------------------------------
+
+with_open_dets(FN) ->
+    DataDir = hpr_utils:base_data_dir(),
+    DETSFile = filename:join([DataDir, "hpr_routes_storage.dets"]),
+    ok = filelib:ensure_dir(DETSFile),
+
+    case
+        dets:open_file(?DETS, [
+            {file, DETSFile}, {type, bag}, {keypos, hpr_route_ets:ets_keypos()}
```
Comment (michaeldjeffrey) [inline src/grpc/iot_config/hpr_route_storage.erl:170]: should be a set

File: src/grpc/iot_config/hpr_route_stream_worker.erl [hunk 3]
```diff
@@ -129,11 +159,40 @@ init(Args) ->
...
         stream = undefined,
-        conn_backoff = Backoff
+        conn_backoff = Backoff,
+        counts = #{
+            route => 0,
+            eui_pair => 0,
+            skf => 0,
+            devaddr_range => 0
+        },
+        %% last_timestamp = persistent_term:get(?MODULE, 0)
```
Comment (michaeldjeffrey) [inline src/grpc/iot_config/hpr_route_stream_worker.erl:179]: remove

File: src/grpc/iot_config/hpr_route_stream_worker.erl [hunk 4]
```diff
@@ -129,11 +159,40 @@ init(Args) ->
     Backoff = backoff:type(backoff:init(?BACKOFF_MIN, ?BACKOFF_MAX), normal),
     lager:info("starting ~p with ~p", [?MODULE, Args]),
     self() ! ?INIT_STREAM,
+
+    ok = open_dets(),
```
Comment (michaeldjeffrey) [inline src/grpc/iot_config/hpr_route_stream_worker.erl:163]: make with_open_dets/1

File: src/grpc/iot_config/hpr_route_stream_worker.erl [hunk 5]
```diff
@@ -166,24 +225,40 @@ handle_call({refresh_route, RouteID}, _From, State) ->
         end,
 
     {reply, Reply, State};
+handle_call(test_counts, _From, State) ->
+    {reply, State#state.counts, State};
+handle_call(test_stream, _From, State) ->
+    {reply, State#state.stream, State};
+handle_call(checkpoint, _From, #state{last_timestamp = LastTimestamp} = State) ->
```
Comment (michaeldjeffrey) [inline src/grpc/iot_config/hpr_route_stream_worker.erl:232]: spawn this

File: src/grpc/iot_config/hpr_route_stream_worker.erl (hunk 4 above)
Comment (michaeldjeffrey) [reply src/grpc/iot_config/hpr_route_stream_worker.erl:163]: The stream worker opens it's own dets 1 time, and closes it with terminate.

File: rebar.config [hunk 6]
```diff
@@ -17,7 +17,7 @@
     {hackney, "1.17.0"},
     {aws, {git, "https://github.com/aws-beam/aws-erlang.git", {tag, "0.7.1"}}},
     {aws_credentials, {git, "https://github.com/aws-beam/aws_credentials.git", {tag, "0.1.9"}}},
-    {helium_proto, {git, "https://github.com/helium/proto.git", {branch, "master"}}},
+    {helium_proto, {git, "https://github.com/helium/proto.git", {branch, "bbalser/iot-config-since"}}},
```
Comment (macpie) [inline rebar.config:20]: Lets try to merge that before?

**Review by macpie — APPROVED** [review]
I am assuming test are fine, they are not running because this is not a PR against main

**Pull request merged**